net/net.go

```diff
@@ -235,13 +236,18 @@ func (n *net) CreateThread(_ context.Context, id thread.ID, opts ...core.NewThre
 }
 
 func (n *net) ensureUnique(id thread.ID) error {
-	_, err := n.store.GetThread(id)
-	if err == nil {
-		return lstore.ErrThreadExists
-	}
+	// Check if thread exists.
+	thrd, err := n.store.GetThread(id)
 	if !errors.Is(err, lstore.ErrThreadNotFound) {
 		return err
 	}
+	if err == lstore.ErrThreadNotFound {
+		return nil
+	}
+	// Ensure we don't already have our 'own' log.
+	if thrd.GetOwnLog().PrivKey != nil {
```

Comment:
GetOwnLog will return a nil ThreadInfo, so we still want to check for PrivKey

Reply:
Could we also check that the logID doesn't already exist? (That would apply to pubkey-only logs)